tests/41end-to-end-keys/07-backup.pl

```diff
@@ -40,6 +38,44 @@
       });
    };
 
+test "Responds correctly when backup is empty",
+   requires => [ $fixture, qw( can_create_backup_version ) ],
+
+   do => sub {
+      my ( $user ) = @_;
+      my $version;
+
+      matrix_get_key_backup_info( $user )->then( sub {
+         my ( $content ) = @_;
+
+         log_if_fail "Content", $content;
+
+         $version = $content->{version};
+
+         matrix_get_backup_key( $user, '!notaroom', 'notassession', $version);
+      })->main::expect_http_4xx
+      ->then( sub {
+         matrix_get_backup_key( $user, '!notaroom', '', $version);
+      })->then( sub {
+         my ( $content ) = @_;
+
+         log_if_fail "Content", $content;
+
+         assert_deeply_eq( $content, {"sessions" => {}});
+
+         matrix_get_backup_key( $user, '', '', $version );
+      })->then( sub {
+         my ( $content ) = @_;
+
+         log_if_fail "Content", $content;
+
+         assert_deeply_eq( $content, {"rooms" => {}});
+
+         Future->done(1);
+         matrix_get_backup_key( $user, '', '', 'bogusversion');
+      })->main::expect_http_404;
```

Comment:
can you use `expect_m_not_found`, to check the error code as well as the http code?